gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -560,6 +567,77 @@ public FSDataInputStream open(Path hadoopPath, int bufferSize) throws IOExceptio
 
     return new FSDataInputStream(in);
   }
+  // Conf not used in the method for sometime, as some changes need to be made for laters
+  private ThreadPoolExecutor initThreadPools() {
+    int maxThreads = DEFAULT_MAX_THREADS.getDefault();
+    if (maxThreads < 2) {
+      logger.atWarning().log("Maximum number of threads should at least be 2.");
+      maxThreads = 2;
+    }
+    int keepAliveTime = KEEP_ALIVE_TIME.getDefault();
+    ThreadPoolExecutor unboundedThreadPool =
+        new ThreadPoolExecutor(
+            maxThreads,
+            Integer.MAX_VALUE,
+            keepAliveTime,
+            TimeUnit.SECONDS,
+            new LinkedBlockingQueue<>(),
+            BlockingThreadPoolExecutorService.newDaemonThreadFactory("gcs-transfer-unbounded"));
+    unboundedThreadPool.allowCoreThreadTimeOut(true);
+    return unboundedThreadPool;
+  }
+
+  public GoogleHadoopFileStatus getGcsFileStatus(FileStatus fileStatus) {
```

Comment:
Why is this method public ? Looking at the usage, does this check need to be a separate method ?

Reply:
I will change it to private. It is required to check if the status is an instance of GcsFileStatus or external file statuses